Wordle.Api/Wordle.Api/Services/LeaderboardService.cs

```diff
@@ -0,0 +1,51 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.IdentityModel.Tokens;
+using Wordle.Api.Data;
+
+namespace Wordle.Api.Services
+{
+    public class LeaderboardService
+    {
+        private readonly AppDbContext _db;
+
+        public LeaderboardService(AppDbContext db)
+        {
+            _db = db;
+        }
+
+        public List<Player> GetLeaderboard()
+        {
+            return  _db.Players.OrderBy(player => player.AverageAttempts).Take(10).ToList();
+        }
+
+        public async Task<Player> AddPlayer(string? name, double Attempts, int secondsPerGame)
+        {
+
+            if (name is null)
+            {
+                throw new ArgumentNullException("Name is required.");
+            }
+
+            var player = await _db.Players.FirstOrDefaultAsync(p => p.Name == name);
+            if (player != null)
+            {
+                player.GameCount += 1;
+                player.AverageAttempts = (player.AverageAttempts + (Attempts - player.AverageAttempts) / player.GameCount);
```

Comment:
Right here i'm not sure why you are calculating the average attempts and average seconds this way. Maybe i'm missing something so feel free to explain your logic to me if that is the case. This is how I went about designing this:
```suggestion
                player.Attempts += Attempts;
                player.AverageAttempts = (player.Attempts / player.GameCount);
```
The way I see it is the players game count, attempts, and total time will continue to increase as they play the game. So to calculate the average we can just divide Attempts (Or Total Time Played) by their Game Count every time they finish a game.

Also I added two more properties to the Player class called TotalAttempts and TotalTimeInSeconds just so I could easily calculate the averages.

Reply:
Yeah this is the source I used to find how to calculate a new average when only given the previous average and the next value. https://math.stackexchange.com/questions/22348/how-to-add-and-subtract-values-from-an-average 
I tested it with a small data set and it works.

Reply:
But why not do what @Nolan-Posey suggested and just re-calculate the average? Why are you trying to add to it?

Reply:
The equation does recalculate the average. I added the link above on how it works. I also tested it with a small set of data to make sure it works. My reasoning for using this was to keep the dto as small as possible.